srv/_modules/cluster.py

```diff
@@ -71,11 +70,7 @@ def storage_device_config():
                 time.sleep(_sleep_time)
 
                 logger.info(f"Command to populate multipath devices: {cmd}")
-                device_list = subprocess.Popen([cmd],
-                                        shell=True,
-                                        stdout=subprocess.PIPE,
-                                        stderr=subprocess.PIPE
-                                    ).stdout.read().decode("utf-8").splitlines()
+                device_list = __utils__['process.simple_process'](cmd)
```

Comment:
@divya-vijay I agree with @ypise : we shouldn't go that path again and `run_subprocess_cmd` is likely generic and good enough to satisfy your need.

But it would bring some other questions:
 - how to import provisioner from within salt env: salt env sometimes may be tricky and may miss some paths in python search list.  That's why [this code](https://github.com/Seagate/cortx-prvsnr/blob/99eef712cdab764f34ef8cd7f2622ee801fea037/srv/_modules/prvsnr.py#L27-L50) was implemented.
 - how to call it, options:
   - call directly from our salt modules
   - call from inside __utils __:process (like you do) so it might be easier to consume by custom modules
   (I think the latter makes sense, it will also hide provisioner import routine)

So I suggest you to try the following options:

- keep new module in utils and put there:
  - provisioner import routine
  - wrapper around provisioner.utils.run_subprocess_cmd (like one that currently call popen)
- call the former from `prvsnr.py` module
- call the latter from all modules that require shell routine

Reply:
@andkononykhin Thank you for this suggestion. This enhances the design better and makes it more structured, I agree. However, for now, since our scope is restricted to only shell commands, I've removed the `srv/_utils` and re-used the existing `run_subprocess_cmd` with some slight modifications to handle the same. And, as we move forward to face more utility functions in `srv/`, I've also added a snippet of this suggestion into the code for others' clarity.